src/main/java/com/epam/lowcost/DAO/UserDAOImpl.java

```diff
@@ -55,6 +57,17 @@ public User getById(long userId) {
         return user;
     }
 
+
+    @Override
+    public User findByEmail(String email) {
+        try {
+            JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+            return jdbcTemplate.queryForObject("SELECT * FROM USERS WHERE email=?", UserRowMapper.getInstance(), email);
```

Comment:
Basically I think you could create the bean of Mapper and inject in the DAO. What do you think about it? Will it better and correct approach ?

Reply:
This is realization we decided to use.

Reply:
Changed to DI usage of Mapper